src/modules/avformat/producer_avformat.c

```diff
@@ -1825,8 +1825,8 @@ static int producer_get_image( mlt_frame frame, uint8_t **buffer, mlt_image_form
 						{
 							// Try to reconnect to live sources by closing context and codecs,
 							// and letting next call to get_frame() reopen.
+							mlt_service_unlock( MLT_PRODUCER_SERVICE( producer ) );
```

Comment:
Since `mlt_service_unlock()` is already called under `exit_get_image:` (L2082), would calling it here be problematic?

Reply:
`prepare_reopen()` immediately calls `mlt_service_lock()` but now I see it also calls `mlt_service_unlock()` at the end. So, I need to relock it here. There is another call to `prepare_reopen()`  inside `producer_get_audio()` which is why that function is not changed. I see I also erroneously removed the unlock on the packets_mutex.